File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
   public List<Integer> preorderTraversal(TreeNode root) {
		return traversal(root,new ArrayList<Integer>());
	}
    
	public List<Integer> traversal(TreeNode root,List<Integer> list){
        if(root==null){
            return list;
        }
        
        list.add(root.val);
        traversal(root.left,list);
        traversal(root.right,list);
      
        return list;
    }
}

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorder(self, node, ans):
        if(node == None):
            return ans;
        
        ans.append(node.val);
        self.preorder(node.left, ans);
        self.preorder(node.right, ans);
        
        return ans;
        
    
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        return self.preorder(root, []);
        